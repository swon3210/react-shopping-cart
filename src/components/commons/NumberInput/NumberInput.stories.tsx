import NumberInput, { Props } from './NumberInput';

export default {
  title: 'Components/Commons/NumberInput',
  component: NumberInput,
  argTypes: {},
};

const Template = (args: Props) => <NumberInput {...args} />;

export const Default = Template.bind({});

(Default as any).args = {};
