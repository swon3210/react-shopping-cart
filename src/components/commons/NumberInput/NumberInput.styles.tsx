import styled from 'styled-components';

export const NumberInput = styled.div`
  display: inline-flex;
  border: 1px solid ${({ theme }) => theme.GRAY_200};
`;

export const Input = styled.input`
  box-sizing: border-box;
  width: 72px;
  height: 60px;
  display: inline;
  font-size: 24px;
  padding: 14px 20px;
  text-align: center;
  border: none;

  &:focus {
    outline: none;
  }
`;

export const TriangleButtonWrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

export const TriangleButton = styled.button`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 42px;
  height: 30px;
  border-left: 1px solid ${({ theme }) => theme.GRAY_200};
  &:not(:last-child) {
    border-bottom: 1px solid ${({ theme }) => theme.GRAY_200};
  }
`;
