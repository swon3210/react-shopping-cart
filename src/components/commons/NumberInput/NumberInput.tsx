import * as Styled from './NumberInput.styles';
import triangleUpSVG from '../../../assets/svgs/triangle-up.svg';
import triangleDownSVG from '../../../assets/svgs/triangle-down.svg';
import React from 'react';

export interface Props extends Pick<React.HTMLAttributes<HTMLInputElement>, 'onChange'> {
  value?: string;
  onIncreaseButtonClick?: () => void;
  onDecreaseButtonClick?: () => void;
}

const NumberInput = ({ value = '1', onChange, onIncreaseButtonClick, onDecreaseButtonClick }: Props) => {
  return (
    <Styled.NumberInput>
      <Styled.Input type="number" value={value} onChange={onChange} />
      <Styled.TriangleButtonWrapper>
        <Styled.TriangleButton onClick={onIncreaseButtonClick}>
          <img src={triangleUpSVG} alt="수량 증가 버튼" />
        </Styled.TriangleButton>
        <Styled.TriangleButton onClick={onDecreaseButtonClick}>
          <img src={triangleDownSVG} alt="수량 감소 버튼" />
        </Styled.TriangleButton>
      </Styled.TriangleButtonWrapper>
    </Styled.NumberInput>
  );
};

export default NumberInput;
