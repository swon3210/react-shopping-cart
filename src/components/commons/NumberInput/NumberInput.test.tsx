import NumberInput from './NumberInput';
import { render } from '@testing-library/react';

describe('NumberInput Component', () => {
  it('NumberInput Snapshot', () => {
    const numberInputUtil = render(<NumberInput />);
    expect(numberInputUtil.asFragment()).toMatchSnapshot();
  });
});
