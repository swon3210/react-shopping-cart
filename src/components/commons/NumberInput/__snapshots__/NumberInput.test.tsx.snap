// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput Component NumberInput Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvaa dfUfZg"
  >
    <input
      class="sc-gsDJrp bUalKf"
      type="number"
      value="1"
    />
    <div
      class="sc-dkPtyc cJFzIl"
    >
      <button
        class="sc-hKwCoD dRYWVj"
      >
        <img
          alt="수량 증가 버튼"
          src="triangle-up.svg"
        />
      </button>
      <button
        class="sc-hKwCoD dRYWVj"
      >
        <img
          alt="수량 감소 버튼"
          src="triangle-down.svg"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
